### PR TITLE
fix: 'generate' vs 'interactive' prompt fixes

### PR DESCRIPTION
**Pull request by mantagen**
## Description

- I had to make some adjustments to the prompt logic for it to work correctly in 'generate' mode

### CODE REVIEW

File: packages/core/src/prompts/lesson-assistant/index.ts [hunk 1]
```diff
@@ -68,7 +69,7 @@ export const getPromptParts = (props: TemplateProps): TemplatePart[] => {
       : undefined,
     americanToBritishSection,
     languageAndVoice,
-    schema,
+    props.isUsingStructuredOutput ? undefined : schema,
```
Comment (mantagen) [inline packages/core/src/prompts/lesson-assistant/index.ts:72]: No need for schema in prompt and response_schema. Should shave off a few tokens.

File: packages/core/src/prompts/lesson-assistant/index.ts [hunk 2]
```diff
@@ -85,5 +86,7 @@ export const generatePromptPartsHash = (props: TemplateProps): string => {
   const parts = getPromptParts(props);
   const partsString = parts.map((part) => part.toString()).join("");
   const hash = crypto.createHash("md5").update(partsString).digest("hex");
-  return `${props.responseMode}-${props.useRag ? "rag" : "noRag"}-${props.baseLessonPlan ? "basedOn" : "notBasedOn"}-${hash}`;
+  return `${props.responseMode}-${props.useRag ? "rag" : "noRag"}-${
```
Comment (mantagen) [inline packages/core/src/prompts/lesson-assistant/index.ts:89]: prettier i guess?

File: packages/aila/src/core/prompt/builders/AilaLessonPromptBuilder.ts [hunk 3]
```diff
@@ -89,6 +89,10 @@ export class AilaLessonPromptBuilder extends AilaPromptBuilder {
         : undefined,
       lessonPlanJsonSchema: JSON.stringify(LessonPlanJsonSchema),
       llmResponseJsonSchema: JSON.stringify(LLMResponseJsonSchema),
+      isUsingStructuredOutput:
```
Comment (mantagen) [inline packages/aila/src/core/prompt/builders/AilaLessonPromptBuilder.ts:92]: could go for the more imperative `useStructuredOutput` for brevity ... or `shouldUse...`

Comment (stefl) [reply packages/aila/src/core/prompt/builders/AilaLessonPromptBuilder.ts:92]: I think we might want to create a prompt variant for this eventually seeing as the prompt does change depending on this variable

File: packages/core/src/prompts/lesson-assistant/index.ts (hunk 1 above)
Comment (stefl) [reply packages/core/src/prompts/lesson-assistant/index.ts:72]: Great - glad we've worked that out

**Review by stefl — APPROVED** [review]

**Pull request merged**